BUG: special: failure of TestSystematic.test_besselj_complex on main

### Describe your issue.

Test failure on Prerelease deps & coverage report, full, py3.10/npMin & py3.11/npPre, dev.py (3.11), e.g. gh-22387, gh-22386.

### Reproducing Code Example

```python
-
```

### Error message

```shell
_____________________ TestSystematic.test_besselj_complex ______________________
[gw0] linux -- Python 3.11.11 /opt/hostedtoolcache/Python/3.11.11/x64/bin/python
scipy/special/tests/test_mpmath.py:798: in test_besselj_complex
    assert_mpmath_equal(
        self       = <scipy.special.tests.test_mpmath.TestSystematic object at 0x7f6c98d25f90>
scipy/special/_mptestutils.py:295: in assert_mpmath_equal
    d.check()
        a          = (<function TestSystematic.test_besselj_complex.<locals>.<lambda> at 0x7f6c17dad620>, <function exception_to_nan.<local....special._mptestutils.Arg object at 0x7f6c171e2ad0>, <scipy.special._mptestutils.ComplexArg object at 0x7f6c13b51c10>])
        d          = <MpmathData: wrap (complex)>
        kw         = {}
scipy/special/_mptestutils.py:282: in check
    raise value
        argarr     = array([[-8.98846567e+307 +0.j        , -1.00000000e+001-10.j        ],
       [-8.98846567e+307 +0.j        , -1.00000...00000e+001 +5.j        ],
       [ 8.98846567e+307 +0.j        ,  1.00000000e+001+10.j        ]],
      shape=(546, 2))
        dps        = 20
        dps_list   = [20]
        j          = 0
        mptype     = <function MpmathData.check.<locals>.mptype at 0x7f6c17dad080>
        old_dps    = 15
        old_prec   = 53
        pytype     = <function mpc2complex at 0x7f6c99c73060>
        self       = <MpmathData: wrap (complex)>
        tb         = <traceback object at 0x7f6c180d3040>
        tp         = <class 'AssertionError'>
        value      = AssertionError('\nMax |adiff|: 0\nMax |rdiff|: 0\nBad results (2 out of 376) for the following points (in output 0):\n...          => -0.-0.j                        != inf+0.j                         (rdiff 0.                            )')
scipy/special/_mptestutils.py:263: in check
    assert_func_equal(
        argarr     = array([[-8.98846567e+307 +0.j        , -1.00000000e+001-10.j        ],
       [-8.98846567e+307 +0.j        , -1.00000...00000e+001 +5.j        ],
       [ 8.98846567e+307 +0.j        ,  1.00000000e+001+10.j        ]],
      shape=(546, 2))
        dps        = 20
        dps_list   = [20]
        j          = 0
        mptype     = <function MpmathData.check.<locals>.mptype at 0x7f6c17dad080>
        old_dps    = 15
        old_prec   = 53
        pytype     = <function mpc2complex at 0x7f6c99c73060>
        self       = <MpmathData: wrap (complex)>
        tb         = <traceback object at 0x7f6c180d3040>
        tp         = <class 'AssertionError'>
        value      = AssertionError('\nMax |adiff|: 0\nMax |rdiff|: 0\nBad results (2 out of 376) for the following points (in output 0):\n...          => -0.-0.j                        != inf+0.j                         (rdiff 0.                            )')
scipy/special/_testutils.py:85: in assert_func_equal
    fdata.check()
E   AssertionError: 
E   Max |adiff|: 0
E   Max |rdiff|: 0
E   Bad results (2 out of 376) for the following points (in output 0):
E   -10.+0.j                        0.+0.j                         => 0.-0.j                         != inf+0.j                         (rdiff 0.                            )
E   -5.+0.j                         0.+0.j                         => -0.-0.j                        != inf+0.j                         (rdiff 0.                            )
        atol       = 1e-300
        data       = array([[-8.98846567e+307 +0.j        , -1.00000000e+001-10.j        ],
       [-8.98846567e+307 +0.j        , -1.00000...00000e+001 +5.j        ],
       [ 8.98846567e+307 +0.j        ,  1.00000000e+001+10.j        ]],
      shape=(546, 2))
        distinguish_nan_and_inf = True
        dtype      = None
        fdata      = <Data for <lambda>>
        func       = <function TestSystematic.test_besselj_complex.<locals>.<lambda> at 0x7f6c17dad620>
        ignore_inf_sign = False
        knownfailure = None
        nan_ok     = True
        nparams    = 2
        param_filter = None
        points     = array([[-8.98846567e+307 +0.j        , -1.00000000e+001-10.j        ],
       [-8.98846567e+307 +0.j        , -1.00000...00000e+001 +5.j        ],
       [ 8.98846567e+307 +0.j        ,  1.00000000e+001+10.j        ]],
      shape=(546, 2))
        result_columns = None
        result_func = <function MpmathData.check.<locals>.<lambda> at 0x7f6c17dad260>
        results    = <function MpmathData.check.<locals>.<lambda> at 0x7f6c17dad260>
        rtol       = 1e-07
        vectorized = False
```

### SciPy/NumPy/Python version and system information

```shell
main
```

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -321,10 +321,7 @@ jobs:
     - name: Install Python packages
       run: |
         python -m pip install cython pythran ninja meson-python pybind11 click rich_click pydevtool
-        python -m pip install --pre --upgrade pytest pytest-cov pytest-xdist gmpy2 threadpoolctl pooch hypothesis matplotlib
-        # Move mpmath back into `install --pre` above once mpmath 1.4.0 with
-        # the fix is out (xref gh-22395)
-        python -m pip install mpmath
+        python -m pip install --pre --upgrade pytest pytest-cov pytest-xdist mpmath gmpy2 threadpoolctl pooch hypothesis matplotlib
         python -m pip install -r requirements/openblas.txt
         # Install numpy last, to ensure we get nightly (avoid possible <2.0 constraints).
         python -m pip install --pre --upgrade --timeout=60 -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy

